### ISSUE TITLE
Is it possible to update demands on a build that is waiting for agent?

### ISSUE THREAD

**Issue by ppejovic** [Microsoft/azure-pipelines-agent#2060]
The TFS/Azure DevOps REST API documentation indicates that you can update the demands on a build and indeed I can invoke the method and get a 200 back. However this appears to have no affect on the build in the queue and it still runs on agents without the capability provided in the demand.

If we have a large number of builds waiting for an agent, we’d like to add a demand to specific builds so they run on a subset of agents, freeing up other agents to run higher priority builds. Is there any way to achieve this?

Thanks

**Comment by TingluoHuang** [comment]
@ppejovic i don't think you can do that, the build is immutable after it get queued. The closest thing i can think about is to disable the definition that has the most of the queued build you don't what them to run now.

**Comment by ppejovic** [comment]
Thanks for the reply, I thought that may be the case. How about paused builds? Is it possible to modify them after being triggered but before the definition is unpaused?

**Comment by TingluoHuang** [comment]
you can't, the build is immutable as long as it get queued (include paused).
